### PR TITLE
userInfo.php data parse

### PR DESCRIPTION
**Pull request by WOSHIZHAZHA120**
Sorry for my english...
probably have some grammatical errors...

### CODE REVIEW

**Comment by SMJSGaming** [comment]
Can you pls order this by id so I can at least review this without having to search the id constantly

**Comment by WOSHIZHAZHA120** [comment]
> Can you pls order this by id so I can at least review this without having to search the id constantly

It's order by id now.

File: docs/resources/server/user.md [hunk 1]
```diff
@@ -1 +1,35 @@
 # User Resource
+| Key | Name/Value | Type | Description |
+| --- | ---------- | ---- | ----------- |
+| 1 | Username | **string** | The name of player |
+| 2 | Userid | **number** | The ID of player |
+| 3 | Stars | **number** | The count of stars player have |
+| 4 | Demons | **number** | The count of demons player have |
```
Comment (SMJSGaming) [inline docs/resources/server/user.md:7]: 6 should be added as another unknown since it does exist

File: docs/resources/server/user.md [hunk 2]
```diff
@@ -1 +1,35 @@
...
+| 23 | accBall | **number** | The ball number of the player use |
+| 24 | accBird | **number** | The bird number of the player use |
+| 25 | accDart(wave) | **number** | The dart(wave) number of the player use |
+| 26 | accRobot | **number** | The robot number of the player use |
+| 28 | accGlow | **number** | The glow number of the player use |
+| 29 | unknown | **number** | unknown but always 1 |
+| 30 | GlobalRank | **number** | The global rank of this player |
+| 31 | Friendstate | **number** | 0: None, 1: already is friend, 3: send request to target, but target haven't accept, 4: target send request, but haven't accept 
+| 43 | accSpider | **number** | The spider number of the player use |
+| 44 | Username | **string** | The name of player |
```
Comment (SMJSGaming) [inline docs/resources/server/user.md]: twitter name not username

File: docs/resources/server/user.md [hunk 3]
```diff
@@ -1 +1,35 @@
...
+| 45 | Username | **string** | The name of player |
```
Comment (SMJSGaming) [inline docs/resources/server/user.md]: twitch name

File: docs/resources/server/user.md [hunk 4]
```diff
@@ -1 +1,35 @@
...
+| 4 | Demons | **number** | The count of demons player have |
+| 8 | Creatorpoints | **number** | The count of creatorpoints player have |
+| 10 | Color1 | **number** | First color of the player use |
+| 11 | Color2 | **number** | Second color of the player use |
+| 13 | Coins | **number** | The count of coins player have |
+| 16 | accountID | **number** | The accountid of this player |
+| 17 | Usercoins | **number** | The count of usercoins player have |
+| 18 | Messagestate | **number** | 0: All, 1: Only friends, 2: None |
+| 19 | FriendsState | **number** | 0: All, 1: None |
+| 20 | Username | **string** | The name of player |
```
Comment (Cvolton) [inline docs/resources/server/user.md:16]: Key 20 is the YouTube channel URL, not the player username

File: docs/resources/server/user.md [hunk 5]
```diff
@@ -1 +1,35 @@
...
+| 29 | unknown | **number** | unknown but always 1 |
+| 30 | GlobalRank | **number** | The global rank of this player |
+| 31 | Friendstate | **number** | 0: None, 1: already is friend, 3: send request to target, but target haven't accept, 4: target send request, but haven't accept 
+| 43 | accSpider | **number** | The spider number of the player use |
+| 44 | Username | **string** | The name of player |
+| 45 | Username | **string** | The name of player |
+| 46 | Diamonds | **number** | The count of diamonds player have |
+| 48 | accExplosion | **number** | The explosion number of the player use |
+| 49 | Modlevel | **number** | 0: None, 1: Normal Mod(yellow), 2: Elder Mod(orange) |
+| 50 | Messagestate | **number** | 0: All, 1: Only friends, 2: None |
```
Comment (SMJSGaming) [inline docs/resources/server/user.md:33]: CommentHistoryState*

**Comment by SMJSGaming** [comment]
A lot which should be fixed

File: docs/resources/server/user.md (hunk 2 above)
Comment (Cvolton) [inline docs/resources/server/user.md:28]: Key #44 is the twitter username

File: docs/resources/server/user.md (hunk 3 above)
Comment (Cvolton) [inline docs/resources/server/user.md:29]: key #45 is the player's twitch username

Comment (WOSHIZHAZHA120) [reply docs/resources/server/user.md]: fixed...
I found my twitter, yt, twitch is same of userName....

**Comment by SMJSGaming** [comment]
Fix my reviews as well since it's not accurate

File: docs/resources/server/user.md (hunk 1 above)
Comment (Cvolton) [reply docs/resources/server/user.md:7]: note that 6 is not always returned, this should be mentioned in the documentation as well

Comment (Cvolton) [reply docs/resources/server/user.md:7]: update: 6 is the global leaderboard position, it's only returned by getGJScores.php with type "relative"

File: docs/resources/server/user.md [hunk 6]
```diff
@@ -1 +1,35 @@
 # User Resource
+| Key | Name/Value | Type | Description |
+| --- | ---------- | ---- | ----------- |
+| 1 | Username | **string** | The name of player |
+| 2 | Userid | **number** | The ID of player |
+| 3 | Stars | **number** | The count of stars player have |
+| 4 | Demons | **number** | The count of demons player have |
+| 8 | Creatorpoints | **number** | The count of creatorpoints player have |
```
Comment (Cvolton) [inline docs/resources/server/user.md:8]: 7 was missed as well, duplicate of 16 (presumably due to historical reasons)

**Comment by WOSHIZHAZHA120** [comment]
many data(like, 6, 7, 16) it's not return in getGJUserInfo.php...

**Comment by SMJSGaming** [comment]
true but it does fall under the category user info responses

**Pull request merged**